pkg/apis/options/secret_source.go

```diff
@@ -0,0 +1,14 @@
+package options
+
+// SecretSource references an individual secret value.
+// Only one source within the struct should be defined at any time.
+type SecretSource struct {
+	// Value expects a base64 encoded string value.
+	Value string `json:"value,omitempty"`
```

Comment:
A value here is just a base64 encoded string, so, isn't particular secret.

Wondering if we should have this/what benefit it actually provides vs users using env vars or values from files 🤔

Reply:
Yes I was wondering the same to be honest. It's like secrets in k8s. Not really secret...

Reply:
tangent. most people don't realize you can actually encrypt k8s secrets on disk. is an api-server setting. So, no, k8s secrets are not just encoded on all clusters, but actually encrypted.

Reply:
On disk inside etcd yes and that should be the default but with cluster access any user with the proper RBAC rolebindings can just decode the secrets but that's beside the point of this PR. So let's not continue this tangent^^